### PR TITLE
Bulk api methods

### PR DESCRIPTION
**Pull request by asedge**
Created some abstractions for creating, running, & seeing the results of Bulk jobs in Salesforce. See the README for usage.

Please note, this code is not production ready yet but it'll give us a chance to experiment with the API and see if it's suitable to speed up some of our API actions.

### CODE REVIEW

**Comment by jacobdaddario** [comment]
It's awesome to see some really great OOP design going on to solve the problems we have with bulk actions in Glue currently 😄

**Comment by asedge** [comment]
[Use bulk methods in ActiveForce for PersistOngoingDepositChange](https://trello.com/c/QQtzc8Kg/2571-use-bulk-methods-in-activeforce-for-persistongoingdepositchange)

File: lib/active_force/bulk/job.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        end
+      end
+
+      # Defines question methods for various states of the job, e.g. #open?, #in_progress?, etc.
+      def initialize_state_methods
+        STATES.values.each do |state|
+          state_method = <<-STATE_METHOD
+            def #{state.to_s.underscore}?
+              @state == '#{state}'
+            end
```
Comment (lazaruslarue) [inline lib/active_force/bulk/job.rb:129]: whoa, neat.

File: README.md [hunk 2]
```diff
@@ -249,6 +249,22 @@ accounts = Account.where(web_enabled: 1).limit(2)
...
+```ruby
+[
+  { id: '11111111', attribute1: 'value1', attribute2: 'value2'},
+  { id: '22222222', attribute1: 'value3', attribute2: 'value4'},
+]
+```
+The attributes will be mapped back to what's expected on the SF side automatically. The response is a `ActiveForce::Bulk::JobResult` object
+which can access `successful` & `failed` results, has some `stats`, and the original `job` object that was used to create and process the
+Bulk job.
+
```
Comment (lazaruslarue) [inline README.md:267]: 👍

**Review by lazaruslarue — COMMENTED** [review]
looks good.

File: README.md [hunk 3]
```diff
@@ -268,4 +284,5 @@ When using rails, you can generate a model with all the fields you have on your
  [1]: http://www.salesforce.com
  [2]: https://github.com/ejholmes/restforce
  [3]: https://github.com/bkeepers/dotenv
+ [4]: https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/bulk_api_2_0.htm
```
Comment (salonishah23) [inline README.md:287]: readme is helpful. I see you've listed the link to the SF Bulk API doc, would it be helpful to explicitly call our the limits of this API here?

Comment (asedge) [reply README.md:287]: I'd rather link to Salesforce as the source of truth than to keep our README up to date with their documentation.

**Pull request merged**